webapp/src/widgets/menu/menu.tsx

```diff
@@ -22,15 +23,46 @@ export default class Menu extends React.PureComponent<Props> {
     static Switch = SwitchOption
     static Separator = SeparatorOption
     static Text = TextOption
+    static TextInput = textInputOption
     static Label = LabelOption
 
+    public state = {
+        hoveringIdx: -1,
+    }
+
     public render(): JSX.Element {
         const {position, children} = this.props
         return (
             <div className={'Menu noselect ' + (position || 'bottom')}>
                 <div className='menu-contents'>
                     <div className='menu-options'>
-                        {children}
+                        {React.Children.map(children, (child, i) => {
```

Comment:
Can we move the content of the map function to another component (can be a functional component inside this file, it is just simpler to understand what is going on, because you can name it.

Reply:
Done.